packages/xstate-inspect/src/index.ts

```diff
@@ -68,80 +56,109 @@ export function createDevTools() {
-        },
-        disconnect: 'pendingConnection'
+export const createInspectMachine = (
+  devTools: XStateDevInterface = globalThis.__xstate__
+) =>
+  createMachine<{
+    client?: any;
+  }>({
+    initial: 'pendingConnection',
+    context: {
+      client: undefined
+    },
+    states: {
+      pendingConnection: {},
+      connected: {
+        on: {
+          'service.state': {
+            actions: (ctx, e) => ctx.client!.send(e)
+          },
+          'service.event': {
+            actions: (ctx, e) => ctx.client!.send(e)
+          },
+          'service.register': {
+            actions: (ctx, e) => ctx.client!.send(e)
+          },
+          'service.stop': {
+            actions: (ctx, e) => ctx.client!.send(e)
+          },
+          'xstate.event': {
+            actions: (_, e) => {
+              const { event } = e;
+              const scxmlEventObject = JSON.parse(event) as SCXML.Event<any>;
+              const service = serviceMap.get(scxmlEventObject.origin!);
+              service?.send(JSON.parse(event));
+            }
+          },
+          unload: {
+            actions: (ctx) => {
+              ctx.client!.send({ type: 'xstate.disconnect' });
+            }
+          },
+          disconnect: 'disconnected'
+        }
+      },
+      disconnected: {
+        type: 'final'
       }
-    }
-  },
-  on: {
-    'xstate.inspecting': {
-      target: '.connected',
-      actions: [
-        assign({ client: (_, e) => e.client }),
-        (ctx) => {
-          globalThis.__xstate__.services.forEach((service) => {
-            ctx.client.send({
-              type: 'service.register',
-              machine: JSON.stringify(service.machine),
-              state: JSON.stringify(service.state || service.initialState),
-              sessionId: service.sessionId
+    },
+    on: {
+      'xstate.inspecting': {
+        target: '.connected',
+        actions: [
+          assign({ client: (_, e) => e.client }),
+          (ctx) => {
+            devTools.services.forEach((service) => {
+              ctx.client.send({
+                type: 'service.register',
+                machine: stringify(service.machine),
+                state: stringify(service.state || service.initialState),
+                sessionId: service.sessionId
+              });
             });
-          });
-        }
-      ]
+          }
+        ]
+      }
     }
-  }
-});
+  });
 
 const defaultInspectorOptions: InspectorOptions = {
   url: 'https://statecharts.io/inspect',
-  iframe: () => document.querySelector<HTMLIFrameElement>('iframe[data-xstate]')
+  iframe: () =>
+    document.querySelector<HTMLIFrameElement>('iframe[data-xstate]'),
+  devTools: () => {
+    const devTools = createDevTools();
+    globalThis.__xstate__ = devTools;
+    return devTools;
+  }
 };
 
+function getLazy<T>(value: MaybeLazy<T>): T {
+  return typeof value === 'function' ? (value as () => T)() : value;
+}
+
+function stringify(value: any): string {
```

Comment:
Couldn't we just use `safeStringify` directly?

Reply:
This is a [recommendation from the `fast-safe-stringify`](https://www.npmjs.com/package/fast-safe-stringify#protip) library itself:

1. Try normal `JSON.stringify`
2. Only if that fails, try `safeStringify`

Reply:
Ok, havent realized that 👍